src/d/actor/d_a_obj_nameplate.cpp

```diff
@@ -1,378 +1,362 @@
-/* 80594754-805947F8 000734 00A4+00 1/1 0/0 0/0 .text            checkHitAngle__16daObjNameplate_cFv
+/* 80594754-805947F8 000734 00A4+00 1/1 0/0 0/0 .text checkHitAngle__16daObjNameplate_cFv
  */
-void daObjNameplate_c::checkHitAngle() {
-    // NONMATCHING
+int daObjNameplate_c::checkHitAngle() {
+    s16 sourceAngle = fopAcM_searchPlayerAngleY(this) - shape_angle.y;
+
+    if (sourceAngle >= 0 && sourceAngle < 0x4000) {
+        return 0;
+    } else if (sourceAngle >= 0x4000 && sourceAngle < 0x7fff) {
+        return 1;
+    } else if (sourceAngle < -0x4000 && sourceAngle >= -0x8000) {
+        return 2;
+    } else if (sourceAngle < 0 && sourceAngle >= -0x4000) {
+        return 3;
+    }
+
+    return -1;
 }
 
-/* 805947F8-80594950 0007D8 0158+00 1/1 0/0 0/0 .text            messageProc__16daObjNameplate_cFv
+/* 805947F8-80594950 0007D8 0158+00 1/1 0/0 0/0 .text messageProc__16daObjNameplate_cFv
  */
 void daObjNameplate_c::messageProc() {
-    // NONMATCHING
+    switch (mShowMessage) {
+    case 0:
+        if (mMessageFlowId != -1 &&
+            cLib_distanceAngleS(shape_angle.GetY(), fopAcM_searchPlayerAngleY(this)) < 0x3000)
+        {
+            fopAcM_OnStatus(this, 0);
+            cLib_onBit<u32>(attention_info.flags, 0x4000000A);
+            attention_info.distances[fopAc_attn_TALK_e] = 0x15;
+            attention_info.distances[fopAc_attn_BATTLE_e] = 0x15;
+            eventInfo.onCondition(1);
+        } else {
+            fopAcM_OffStatus(this, 0);
+            cLib_offBit<u32>(attention_info.flags, 0x4000000A);
+        }
+
+        if (dComIfGp_event_runCheck() && eventInfo.checkCommandTalk()) {
+            mMsgFlow.init(this, mMessageFlowId, 0, 0);
+            mShowMessage = 1;
+        }
+        break;
+
+    case 1:
+        dComIfG_play_c& play = g_dComIfG_gameInfo.play;
+        if (play.getEvent().runCheck() && eventInfo.checkCommandTalk() != 0 &&
+            mMsgFlow.doFlow(this, NULL, 0) != 0)
+        {
+            play.getEvent().reset();
+            mShowMessage = 0;
+        }
+        break;
+    }
+
+    return;
 }
 
-/* ############################################################################################## */
-/* 80594F0C-80594F0C 000098 0000+00 0/0 0/0 0/0 .rodata          @stringBase0 */
-#pragma push
-#pragma force_active on
-SECTION_DEAD static char const* const stringBase_80594F0C = "J_Hyosatu";
-SECTION_DEAD static char const* const stringBase_80594F16 = "J_Hyousatu.bmd";
-#pragma pop
-
 /* 80594F28-80594F2C -00001 0004+00 3/3 0/0 0/0 .data            l_arcName */
-SECTION_DATA static void* l_arcName = (void*)&d_a_obj_nameplate__stringBase0;
+static char* l_arcName = "J_Hyosatu";
 
-/* 80594950-805949C8 000930 0078+00 1/1 0/0 0/0 .text            createSolidHeap__FP10fopAc_ac_c */
-static void createSolidHeap(fopAc_ac_c* param_0) {
-    // NONMATCHING
-}
+int daObjNameplate_c::createHeap() {
+    J3DModelData* modelData = (J3DModelData*)dComIfG_getObjectRes(l_arcName, "J_Hyosatu.bmd");
```

Comment:
There is actually a different string here... 
```suggestion
    J3DModelData* modelData = (J3DModelData*)dComIfG_getObjectRes(l_arcName, "J_Hyousatu.bmd");
```